src/librustc_trans/mir/mod.rs

```diff
@@ -151,15 +150,15 @@ impl<'blk, 'tcx> MirContext<'blk, 'tcx> {
     // it may so happen that the current span belongs to a different file than the DIScope
     // corresponding to span's containing visibility scope.  If so, we need to create a DIScope
     // "extension" into that file.
-    fn scope_metadata_for_span(&self, scope_id: mir::VisibilityScope, span: Span)
+    fn scope_metadata_for_loc(&self, scope_id: mir::VisibilityScope, pos: BytePos)
                                -> llvm::debuginfo::DIScope {
         let scope_metadata = self.scopes[scope_id].scope_metadata;
-        if span.lo < self.scopes[scope_id].start_pos ||
-           span.lo > self.scopes[scope_id].end_pos {
+        if pos < self.scopes[scope_id].file_start_pos ||
+           pos > self.scopes[scope_id].file_end_pos {
```

Comment:
This should be `>=` since `file_end_pos` is the first position that's not in the file anymore.